### ISSUE TITLE
BLE module improvement and bug fixing on Thunkable X

### ISSUE THREAD

**Issue by amontefusco** [thunkable/thunkable-issues#328]
**Is your feature request related to a problem? Please describe.**
BLE module.
When compared with the one provided in App Inventor2 is simply…discouraging.
List of improvement (as seen on Android 8.0.0):

1.     speed up the scan operation is slow and very unreliable
2.     if BT is turned off, an error is returned (instead of a popup to the user)
3.     no event management!
4.     no Disconnect method !
5.     no way to specify service_id
6.     no support for service advertisement
7.     no parameters configurable

**Describe the solution you'd like**
The Scan has to be quick and reliable (I have another app written with App Inv 2 that is blazingly quick).

**Describe alternatives you've considered**
MIT App Inventory 2. I arrived to Thunkable due to the iOS support.  

**Which platform are you asking for?**
Thunkable ✕ (x.thunkable.com)

**Additional context**
Android 8.0.0 on Samsung A6 2017.

**Comment by mark-friedman** [comment]
Thanks, @amontefusco!

**Comment by gre4twh1te** [comment]
I agree - the current BLE component is pretty useless.. I cannot convert/submit HEX values that are needed by most of my BLE devices, no disconnect, no connection state,... I hope this is added because right now you cannot create anything meaningful with this component.

**Comment by AnatiliyKsl** [comment]
Hello ! I would like to hear from the developers WHEN WILL THERE BE A FULL BLE CLASS? And BLUETOOTH?

**Comment by anttok** [comment]
I agree, I came for the cross platform support, but i can't stay without full BLE support. Is this still in development, its pretty slow and unreliable at the moment.

**Comment by domhnallohanlon** [comment]
Hey @anttok can you tell me a little more about what you're trying to do via BLE? 
Thanks
